### PR TITLE
Replace nose by pytest

### LINKED ISSUE #498: Stop using nose for testing
Nose is [not actively developed ](https://nose.readthedocs.io/en/latest/index.html)anymore.

We should switch to another, more modern, testing framework. I think pytest is the goto solution right now.

### PR DESCRIPTION
**Pull request by pgajdos**
Fixes #498 

Use unittest only. nose is deprecated and will stop functioning in near future python version.

### CODE REVIEW

**Comment by pgajdos** [comment]
If I should add something to CHANGELOG.md, let me know.

File: tests/test_converter.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
-from nose.tools import assert_equal
+import unittest
```
Comment (pietermarsman) [inline tests/test_converter.py:1]: I prefer to use pytest for everything, and not use a mix of unittest and pytest. Could you pytest'ify all the tests? 

E.g. many tests can be parametrized. And a lot of tests don't need to be grouped in a class anymore.

I'm ok with a proper rewrite/cleanup of the tests.

**Review by pietermarsman — CHANGES_REQUESTED** [review]

**Comment by pietermarsman** [comment]
@pgajdos do you have time to continue to work on this in the near future?

**Comment by pgajdos** [comment]
Apologize, I have missed notification.

@pietermarsman, reading comment from Sep 12, I will probably rather leave it to you.

**Comment by pietermarsman** [comment]
Ok. Thanks for the effort!

**Pull request closed without merging**